rskj-core/src/test/java/co/rsk/mine/MinerServerTest.java

```diff
@@ -763,6 +778,68 @@ public void secondBuildBlockToMineTurnsNotifyFlagOff() {
         }
     }
 
+    @Test
+    public void extraDataNotInitializedWithClientData() {
+        MinerServer minerServer = new MinerServerImpl(
+                config,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                new BuildInfo("cb7f28e", "master"),
+                ConfigUtils.getDefaultMiningConfig()
+        );
+
+        byte[] extraData = minerServer.getExtraData();
+        RLPList decodedExtraData = RLP.decodeList(extraData);
+        assertEquals(2, decodedExtraData.size());
+
+        byte[] firstItem = decodedExtraData.get(0).getRLPData();
+        assertNotNull(firstItem);
+        assertEquals(1, (RLP.decodeInt(firstItem,0)));
+
+        byte[] secondItem = decodedExtraData.get(1).getRLPData();
+        assertNotNull(secondItem);
+        assertEquals("SNAPSHOT-cb7f28e", new String(secondItem));
+    }
+
+    @Test
+    public void extraDataWithClientData() {
+        MinerServer minerServer = new MinerServerImpl(
+                config,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                new BuildInfo("cb7f28e", "master"),
+                ConfigUtils.getDefaultMiningConfig()
+        );
+
+        minerServer.setExtraData("tincho".getBytes());
+
+        byte[] extraData = minerServer.getExtraData();
+        RLPList decodedExtraData = RLP.decodeList(extraData);
+        assertEquals(3, decodedExtraData.size());
+
+        byte[] firstItem = decodedExtraData.get(0).getRLPData();
+        assertNotNull(firstItem);
+        assertEquals(1, (RLP.decodeInt(firstItem,0)));
+
+        byte[] secondItem = decodedExtraData.get(1).getRLPData();
+        assertNotNull(secondItem);
+        assertEquals("SNAPSHOT-cb7f28e", new String(secondItem));
+
+        byte[] thirdItem = decodedExtraData.get(2).getRLPData();
+        assertNotNull(thirdItem);
+        assertEquals("tincho", new String(thirdItem));
+    }
+
     private BtcBlock getMergedMiningBlockWithOnlyCoinbase(MinerWork work) {
         return getMergedMiningBlock(work, Collections.emptyList());
     }
```

Comment:
I think also we can add a test to after a build a block check that the extra data is there.

Reply:
Sounds good, will add this to the backlog for a next release since it's not major and we need to advance with merging this PR soon.